test/jdk/javax/swing/text/html/CSS/CSSAttributeEqualityBug.java

```diff
@@ -0,0 +1,198 @@
+
+    public static void main(String[] args) {
+        testFontSize();
+        testFontFamily();
+        testFontWeight();
+        testColor();
+        testBorderStyle();
+        testCSSLengthValue();
+        testBackgroundPosition();
+        testCSSStringValue();
+    }
+
+    private static void testFontSize() {
+        StyleSheet ss = new StyleSheet();
+        String fontSize = "42";
+
+        SimpleAttributeSet a = new SimpleAttributeSet();
+        ss.addCSSAttribute(a, CSS.Attribute.FONT_SIZE, fontSize);
+
+        SimpleAttributeSet b = new SimpleAttributeSet();
+        ss.addCSSAttribute(b, CSS.Attribute.FONT_SIZE, fontSize);
+
+        if (a.isEqual(b)) {
+            System.out.println( "a equals b");
+        } else {
+            System.out.println("a = " + a);
+            System.out.println("b = " + b);
+            throw new RuntimeException("CSS.Attribute.FontSize a is not equal to b");
+        }
+    }
+
+    private static void testFontFamily() {
+        StyleSheet ss = new StyleSheet();
+        String fontFamily = "Sans-Serif";
+
+        SimpleAttributeSet a = new SimpleAttributeSet();
+        ss.addCSSAttribute(a, CSS.Attribute.FONT_FAMILY, "Sans-Serif");
+
+        SimpleAttributeSet b = new SimpleAttributeSet();
+        ss.addCSSAttribute(b, CSS.Attribute.FONT_FAMILY, "Sans-Serif");
+
+        if (a.isEqual(b)) {
+            System.out.println( "a equals b");
+        } else {
+            System.out.println("a = " + a);
+            System.out.println("b = " + b);
+            throw new RuntimeException("CSS.Attribute.FontFamily a is not equal to b");
+        }
+    }
+
+    private static void testFontWeight() {
+        StyleSheet ss = new StyleSheet();
+        String fontWeight = "bold";
+
+        SimpleAttributeSet a = new SimpleAttributeSet();
+        ss.addCSSAttribute(a, CSS.Attribute.FONT_WEIGHT, fontWeight);
+
+        SimpleAttributeSet b = new SimpleAttributeSet();
+        ss.addCSSAttribute(b, CSS.Attribute.FONT_WEIGHT, fontWeight);
+
+        if (a.isEqual(b)) {
+            System.out.println( "a equals b");
+        } else {
+            System.out.println("a = " + a);
+            System.out.println("b = " + b);
+            throw new RuntimeException("CSS.Attribute.FontWeight a is not equal to b");
+        }
+    }
+
+    private static void testColor() {
+        StyleSheet ss = new StyleSheet();
+        String fontColor = "red";
+
+        SimpleAttributeSet a = new SimpleAttributeSet();
+        ss.addCSSAttribute(a, CSS.Attribute.COLOR, fontColor);
+
+        SimpleAttributeSet b = new SimpleAttributeSet();
+        ss.addCSSAttribute(b, CSS.Attribute.COLOR, fontColor);
+
+        if (a.isEqual(b)) {
+            System.out.println( "a equals b");
+        } else {
+            System.out.println("a = " + a);
+            System.out.println("b = " + b);
+            throw new RuntimeException("CSS.Attribute.Color a is not equal to b");
+        }
```

Comment:
Does it make sense to introduce a new method which compares the two attributes sets `a` and `b`, prints messages and throws the exception? You'll need to pass the CSS attribute name to it. The code will be cleaner.

Reply:
ok